workflow-runtime/src/commonMain/kotlin/com/squareup/workflow1/RenderWorkflow.kt

```diff
@@ -166,14 +161,13 @@ public fun <PropsT, OutputT, RenderingT> renderWorkflowIn(
       // we don't surprise anyone with an unexpected rendering pass. Show's over, go home.
       if (!isActive) return@launch
 
-      // If the action did produce an Output, we send it immediately after the render pass.
-      nextRenderAndSnapshot = renderAndEmitOutput(runner, actionResult, onOutput)
+      nextRenderAndSnapshot = runner.nextRendering()
 
       if (runtimeConfig == ConflateStaleRenderings) {
-        // With this runtime modification, we do not pass renderings we know to be stale. This
-        // means that we may be calling onOutput out of sync with the update of the UI. Output
-        // is an event though, and should always occur immediately - i.e. it cannot be stale.
-        while (actionResult != ActionsExhausted) {
+        // Only null will allow us to continue processing actions and conflating stale renderings.
+        // If this is not null, then we had an Output and we want to send it with the Rendering
+        // (stale or not).
+        while (actionResult == null) {
```

Comment:
PTAL again @rjrjr as I changed the logic here such that if we ever had an Output we would emit the rendering and the Output so that we always satisfy the same constraints.

Reply:
Do we still need the enum? Seems like we're checking `is WorkflowOutput` at this point.

Reply:
Good point. Though we may go back there. I'll leave the slight YAGNI in for now.